migrate/migrate.go

```diff
@@ -22,6 +22,27 @@ import (
 	"github.com/scylladb/gocqlx/qb"
 )
 
+// AwaitSchemaAgreementOption controls whether checking for cluster schema agreement
+// is disabled or if it is checked before each file or statement is applied.
+var AwaitSchemaAgreementOption awaitSchemaAgreement
+
+type awaitSchemaAgreement int
+
+// Options for checking schema agreement.
+const (
+	AwaitSchemaAgreementDisabled awaitSchemaAgreement = iota
+	AwaitSchemaAgreementBeforeEachFile
+	AwaitSchemaAgreementBeforeEachStatement
+	awaitSchemaAgreementAfterMigrations
+)
+
+func (as awaitSchemaAgreement) Check(ctx context.Context, session *gocql.Session, stage awaitSchemaAgreement) error {
```

Comment:
Docs

Reply:
In my view it would be better to separate the fileterig (SchouldAwait) logic from the `session.AwaitSchemaAgreement` call.

Reply:
Yeah, sorry about that – done.

Reply:
Oh damn, didn't see your second comment there.

Reply:
Not sure what you mean I should change. Can you elaborate? 😅

Reply:
Take out `session.AwaitSchemaAgreement(ctx)` from this function and rename to `SchouldAwait`.

Reply:
All done. Take another look? I liked the previous version better because it saved having to use 2 nested `if`s but it's your call. :)